pages/emotion-records.js

```diff
@@ -84,50 +249,116 @@ export default function EmotionRecords({
     const results = fuse.search(searchTerm);
     const items = results.map((result) => result.item);
     setShownEntries(items);
-  }, [emotionEntries, searchTerm]);
-
-  function handleShowHighlighted() {
-    setIsHighlighted(!isHighlighted);
-  }
+  }, [searchTerm, filteredEntries]);
 
   function handleSearch(input) {
     setSearchTerm(input);
   }
 
+  function getFormattedDate(selectedDate) {
+    const date = new Intl.DateTimeFormat(`de`, {
+      day: "2-digit",
+      month: "2-digit",
+      year: "numeric",
+    }).format(selectedDate);
+
+    return date;
+  }
+
   return (
     <StyledFlexColumnWrapper>
       <StyledPageHeader>
```

Comment:
The name StyledPageHeader seems a bit misleading to me. You already have a Header component in the layout. The name (Styled)FilterSection seems more appropriate to me from what code it contains.